data/seed/Seeding.md

```diff
@@ -0,0 +1,180 @@
+- Vite front end server running in another terminal window.
+- Django development server running in another terminal window.
+
+At this point, you should be able to login to the Admin website at
+http://localhost:8000/ or http:127.0.0.1:8000/ with the email address and
+password of the super user you created.
+
+In a fourth terminal window, have your Python virtual environment activated and
+be in your working directory with the UIPA.org source files cloned from your
+fork of the main branch of the [UIPA.org
+repository](https://github.com/CodeWithAloha/uipa).
+
+## Seeding
+
+### Load Classifications, Jurisdictions, and Freedom of Information Laws
+
+Three fixtures will be used to load seed data that is not going to change very
+much for development work.
+
+This will set up the data that needs to exist in the database before you upload
+the public body information from a CSV file.
+
+```
+$ python manage.py loaddata data/seed/2024-03-15-classification.json
+$ python manage.py loaddata data/seed/2024-03-15-jurisdiction.json
+$ python manage.py loaddata data/seed/2024-03-15-foilaw.json
+```
+
+*NOTE:*
+
+> The Classifications data has been extracted from the Classifications field in
+> the extract of public bodies from the production UIPA.org website on
+> 03/15/2024. See 2024-03-15-Hawaii_UIPA_Public_Bodies_All.csv.
+
+
+### First Test Load of Public Bodies
+
+For the first time that you seed the database, you'll be using a very small
+subset of data to see how things go.
+
+- Run this command to load the category data that's needed to load the subset
+  of public body data:
+
+    ```
+    $ python manage.py loaddata data/seed/test-categories.json
+    ```
+
+- Then, on the `Public Body` page (Home > Public Body > Public Bodies), scroll
+  down to the bottom of the page to where there is a `Choose File` button next
+  to the `Import Public Bodies` button.
+
+- Click on the `Choose File` button and navigate to the directory with
+  `test-public-bodies.csv` file. Select that file and click on the `Open`
+  button.
+
+- Back on the `Public Body` page, click on the `Import Public Bodies` button.
+
+- If this works, you should see a message on the `Public Body` page that says
+  that the public bodies were imported.
+
+### Loading all public bodies
+
+If the first test loading of the database with the very small set of data works,
+you can load the full set of public bodies.
+
+- Stop your Django server by pressing Ctl-C in the terminal window running the
+  server.
+
+- Run these commands to re-initialize the database:
+
+    ```
+    $ sh data/seed/clear_db.sh
+    $ sh data/seed/init_db.sh
+    ```
+
+- To load a full set of data, you can use these files:
+    - 2024-03-24-categories.json
+    - 2024-03-24-public-bodies-fixed.csv
+
+- Load the categories:
+    ```
+    $ python manage.py loaddata data/seed/2024-03-24-categories.json
+    ```
+- On the Public Bodies page of the Admin website, upload the
```

Comment:
Weirdly the previous `import_csv` command fails here:

```
➜  uipa git:(pr/80) python manage.py import_csv data/seed/2024-03-24-public-bodies-fixed.csv
/Users/tylerchong/.pyenv/versions/3.10.13/lib/python3.10/site-packages/django_elasticsearch_dsl/documents.py:178: ElasticsearchWarning: Elasticsearch built-in security features are not enabled. Without authentication, your cluster could be accessible to anyone. See https://www.elastic.co/guide/en/elasticsearch/reference/7.15/security-minimal-setup.html to enable security.
  response = bulk(client=self._get_connection(), actions=actions, **kwargs)
Traceback (most recent call last):
  File "/Users/tylerchong/Desktop/workspace/codewithaloha/uipa/manage.py", line 13, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/tylerchong/.pyenv/versions/3.10.13/lib/python3.10/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/Users/tylerchong/.pyenv/versions/3.10.13/lib/python3.10/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/tylerchong/.pyenv/versions/3.10.13/lib/python3.10/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/tylerchong/.pyenv/versions/3.10.13/lib/python3.10/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/Users/tylerchong/Desktop/workspace/codewithaloha/uipa/src/froide/froide/publicbody/management/commands/import_csv.py", line 25, in handle
    importer.import_from_file(f)
  File "/Users/tylerchong/Desktop/workspace/codewithaloha/uipa/src/froide/froide/publicbody/csv_import.py", line 47, in import_from_file
    self.import_row(row)
  File "/Users/tylerchong/Desktop/workspace/codewithaloha/uipa/src/froide/froide/publicbody/csv_import.py", line 90, in import_row
    row["parent"] = PublicBody._default_manager.get(slug=slugify(parent))
  File "/Users/tylerchong/.pyenv/versions/3.10.13/lib/python3.10/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/Users/tylerchong/.pyenv/versions/3.10.13/lib/python3.10/site-packages/django/db/models/query.py", line 637, in get
    raise self.model.DoesNotExist(
froide.publicbody.models.PublicBody.DoesNotExist: PublicBody matching query does not exist.
```

but going through the Web UI with the same file seems to work, so :shrug:

![Screenshot 2024-03-25 at 9 31 41 PM](https://github.com/CodeWithAloha/uipa/assets/15609358/e33d3290-a8f0-4f30-8e37-a892bab51d92)

Reply:
I'm not quite sure why it does that. I get the *PublicBody matching query does not exist." message when I upload via the Admin website. I get 113 public bodies loaded but the CSV file has 201.

Reply:
The error message is saying that the parent of the public body being loaded doesn't exist in the Public Body table yet. So things stop at this point. It's a data issue. I'll need to clean up the data a bit.

Reply:
👍 - I'll merge in https://github.com/CodeWithAloha/uipa/pull/81 then in the meantime

Reply:
The  `data/seed/2024-03-24-public-bodies-fixed.csv` file appears to be good because after several attempts all 201 public bodies are loaded.

In my working directory, there appears to be something flaky going on with `src/froide/froide/publicbody/csv_import.py` in my virtual environment. If I add a few print statements all 201 public bodies will load. I'm using Python 3.10.13 in a virtual environment created using the venv module. When only 118 or so public bodies are loaded, it usually due to the name and the slug not matching up. This prevents public bodies with parent bodies to fail to load.